Octokit.Tests.Integration/EnterpriseHelper.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Octokit.Tests.Integration
+{
+    public static class EnterpriseHelper
+    {
+        static readonly Lazy<Credentials> _credentialsThunk = new Lazy<Credentials>(() =>
+        {
+            var githubUsername = Environment.GetEnvironmentVariable("OCTOKIT_GHE_USERNAME");
+            GHEUserName = githubUsername;
+            GHEOrganization = Environment.GetEnvironmentVariable("OCTOKIT_GHE_ORGANIZATION");
+
+            var githubToken = Environment.GetEnvironmentVariable("OCTOKIT_GHE_OAUTHTOKEN");
+
+            if (githubToken != null)
+                return new Credentials(githubToken);
+
+            var githubPassword = Environment.GetEnvironmentVariable("OCTOKIT_GHE_PASSWORD");
+
+            if (githubUsername == null || githubPassword == null)
+                return null;
+
+            return new Credentials(githubUsername, githubPassword);
+        });
+
+        static readonly Lazy<Credentials> _oauthApplicationCredentials = new Lazy<Credentials>(() =>
+        {
+            var applicationClientId = ClientId;
+            var applicationClientSecret = ClientSecret;
+
+            if (applicationClientId == null || applicationClientSecret == null)
+                return null;
+
+            return new Credentials(applicationClientId, applicationClientSecret);
+        });
+
+        static readonly Lazy<Credentials> _basicAuthCredentials = new Lazy<Credentials>(() =>
+        {
+            var githubUsername = Environment.GetEnvironmentVariable("OCTOKIT_GHE_USERNAME");
+            GHEUserName = githubUsername;
+            GHEOrganization = Environment.GetEnvironmentVariable("OCTOKIT_GHE_ORGANIZATION");
+
+            var githubPassword = Environment.GetEnvironmentVariable("OCTOKIT_GHE_PASSWORD");
+
+            if (githubUsername == null || githubPassword == null)
+                return null;
+
+            return new Credentials(githubUsername, githubPassword);
+        });
+
+        static readonly Lazy<bool> _gitHubEnterpriseEnabled = new Lazy<bool>(() =>
+        {
+            string enabled = Environment.GetEnvironmentVariable("OCTOKIT_GHE_ENABLED");
+            return !String.IsNullOrWhiteSpace(enabled);
+        });
+        
+        static readonly Lazy<Uri> _gitHubEnterpriseUrl = new Lazy<Uri>(() =>
+        {
+            string uri = Environment.GetEnvironmentVariable("OCTOKIT_GHE_URL");
+
+            if (uri != null)
+                return new Uri(uri);
+
+            return null;
+        });
+
+        static EnterpriseHelper()
+        {
+            // Force reading of environment variables.
+            // This wasn't happening if UserName/Organization were 
+            // retrieved before Credentials.
+            Debug.WriteIf(GHECredentials == null, "No credentials specified.");
+        }
+
+        public static string GHEUserName { get; private set; }
```

Comment:
:grey_question: rather than the abbreviation, would it be worth just prefixing these properties with `Enterprise` to make this a bit more readable? Or is that redundant given we're already in `EnterpriseHelper`...